### ISSUE TITLE
False positive: reportUnreachable triggered incorrectly with if a is None or b is None

### ISSUE THREAD

**Issue by wangvation** [DetachHead/basedpyright#1479]
### Description

When using base-pyright, I encountered a false positive with reportUnreachable.

Example code:
```python
def content_similarity(a: str, b: str):
    if a == b:
        return 1.0

    if a is None or b is None:
        return 0.0
```

base-pyright reports that the line return 0.0 is unreachable.
This is not correct. For example:

Case a = "abc", b = None:

First condition a == b → False

Second condition a is None or b is None → True

The code does reach return 0.0.

So the linter’s analysis is overly aggressive and assumes that the first condition covers all cases, which it doesn’t.

Expected Behavior

The return 0.0 should not be flagged as unreachable, because the condition a is None or b is None can be true when a == b is false.

Environment

VS Code version: [1.99.3]

base-pyright version: [v1.31.3]

Python version: [3.12.3]

OS: [Linux Mint 22.1]

<img width="1129" height="194" alt="Image" src="https://github.com/user-attachments/assets/9eeefe7e-f1e1-4003-91ca-d4e86d355ceb" />

<img width="1434" height="397" alt="Image" src="https://github.com/user-attachments/assets/c00465bb-5134-4b27-8004-9c7a9859e787" />

**Comment by KotlinIsland** [comment]
https://basedpyright.com/?typeCheckingMode=all&code=CYUwZgBAxg9gdgFxIg%2BgZwJYFsMBsCGAThggJ4AU%2BAXBGgoQDQQBGNdhAlFQFAR8QZI%2BCAF4RLHvykRCIBAFdCcCAEYAdAAZuvfoIjCMaCADl4ICDEIsBR03BCTpfWQqUQNm7bLTzcCUdDwSKiYOATEZOR2IEzRHNxAA

**Comment by KotlinIsland** [comment]
it's reported as unreachable because `a` and `b` can never be `None`, only `str`, if you expect people to pass `None`, then you need to make your type annotation`str | None`

**Comment by wangvation** [comment]
Oh,I see. Thank you!
